actionview/lib/action_view/lookup_context.rb

```diff
@@ -132,6 +132,11 @@ def exists?(name, prefixes = [], partial = false, keys = [], **options)
       end
       alias :template_exists? :exists?
 
+      def template_exists_in_any_format?(name, prefixes = [], partial = false, keys = [], **options)
+        options_with_all_formats = options.merge(formats: default_formats)
+        @view_paths.exists?(*args_for_lookup(name, prefixes, partial, keys, options_with_all_formats))
+      end
```

Comment:
I also don't know what the semantics should be for variants, do you have thoughts about this?

(It also seems good to enumerate the rules in a comment somewhere, by the way: https://github.com/rails/rails/issues/20666#issuecomment-116882771)